File: rd-ms-usermanagement-client/src/main/java/io/radien/ms/usermanagement/client/entities/Page.java
package io.radien.ms.usermanagement.client.entities;

import java.io.Serializable;
import java.util.List;

public class Page<T extends Serializable> implements Serializable {

    private static final long serialVersionUID = 6812608123262067424L;
    private List<T> results;
    private int currentPage;
    private int totalResults;
    private int totalPages;

    public Page(List<T> results, int currentPage, int totalResults, int totalPages) {
        this.results = results;
        this.currentPage = currentPage;
        this.totalResults = totalResults;
        this.totalPages = totalPages;
    }

    public List<T> getResults() {
        return results;
    }

    public void setResults(List<T> results) {
        this.results = results;
    }

    public int getCurrentPage() {
        return currentPage;
    }

    public void setCurrentPage(int currentPage) {
        this.currentPage = currentPage;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(int totalResults) {
        this.totalResults = totalResults;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(int totalPages) {
        this.totalPages = totalPages;
    }
}

File: front-end-workspace/protec-angular-workspace/projects/protec/src/app/shared/models/page/Page.ts
interface IPage<ObjectType> {
  currentPage: number,
  results: ObjectType[] | undefined,
  totalPages: number,
  totalResults: number,
}

export default class Page<Model> implements IPage<Model> {
  currentPage: number;
  results: Model[];
  totalPages: number;
  totalResults: number;

  constructor(currentPage: number, totalPages: number, totalResults: number, results: Model[]) {
    this.currentPage = currentPage;
    this.totalPages = totalPages;
    this.totalResults = totalResults;
    this.results = results;
  }
} 
